apply action: a11y comment exceeds GitHub's 65,536-char limit for large-preview modules

## Symptom

The `apply` composite action's render check fails at the final "post a11y comment" step for modules with many previews:

```
GraphQL: Body is too long (maximum is 65536 characters) (addComment)
##[error]Process completed with exit code 1.
```

The renders themselves succeed (the preview-diff comment posts fine) — only the a11y comment overflows GitHub's per-comment 65,536-character API cap, which then reds the whole check.

## Where

- `.github/actions/apply/pipelines/a11y.sh` builds the a11y comment body (`_a11y_comment.md`) with no length bound.
- `.github/actions/apply/lib/post-comment.sh` posts it via `gh pr comment --body-file` / `gh api … body=@file`. It already guards **ARG_MAX** (~128 KB, "Argument list too long") by streaming from a file, but there's **no guard for the GitHub comment-API limit (65,536 chars)** — a separate ceiling that streaming doesn't help with.

So any module whose a11y report renders to >65,536 chars hard-fails the check.

## Trigger / evidence

`yschimke/homeassistant-remotecompose` `:previews` (~287 previews) overflows it. First hit on PR #420 there:
- run https://github.com/yschimke/homeassistant-remotecompose/actions/runs/28509006374
- job https://github.com/yschimke/homeassistant-remotecompose/actions/runs/28509006374/job/84504422762 (via the pinned `yschimke/compose-ai-tools/.github/actions/apply@v0.16.8`)

It's a scaling issue, not specific to any one PR — it will trip for any sufficiently large module and gets worse as previews are added.

## Suggested fix

Cap the a11y comment to fit the API limit, in `post-comment.sh` (defence in depth for every comment it posts) and/or when `a11y.sh` assembles the body:

- Keep the summary + highest-severity findings; collapse the rest under a `<details>` and/or append a `… and N more findings` footer, truncating to a safe budget (e.g. ~60 KB to leave headroom).
- Alternatively, split across multiple comments, or upload the full a11y report as a workflow artifact / to the render branch and link to it from a short comment.

A generic guard in `post-comment.sh` (truncate `BODY_FILE` to N bytes on a line boundary + append a truncation notice) would fix this for the a11y comment and any other large body the action posts.

Context: surfaced while publishing the HA design catalog; the design-artifacts pipeline is unaffected (it posts no comment), and the actual renders/semantics were fine — this only breaks the PR-comment check.

--- a/.github/actions/apply/action.yml
+++ b/.github/actions/apply/action.yml
@@ -653,15 +653,15 @@ runs:
         # resolved body, no-diff+no-prior → only post resolved if
         # comment-on-empty-diff=true.
         RESOLVED_BODY="${MARKER}"$'\n'"## Resource Changes"$'\n\n'"_No resource changes in this PR's latest render._"
-        # Stream body via file when present — large resource diffs would
-        # otherwise blow ARG_MAX. RESOLVED_BODY is small and stays inline.
+        # Route the populated-body path through post-comment.sh so a large
+        # resource diff gets the same ARG_MAX streaming *and* the
+        # 65,536-char truncation guard as every other comment (the helper
+        # re-resolves the sticky comment by MARKER and PATCHes/posts).
+        # RESOLVED_BODY is small and stays inline.
         if [ -s _comment_body_resources.md ]; then
-          if [ -n "$COMMENT_ID" ]; then
-            gh api "repos/${REPO}/issues/comments/${COMMENT_ID}" \
-              -X PATCH -f "body=@_comment_body_resources.md"
-          else
-            gh pr comment "$PR_NUMBER" --body-file _comment_body_resources.md
-          fi
+          export PR_NUMBER MARKER
+          export BODY_FILE="_comment_body_resources.md"
+          bash "$ACTION_PATH/lib/post-comment.sh"
         elif [ -n "$COMMENT_ID" ]; then
           gh api "repos/${REPO}/issues/comments/${COMMENT_ID}" \
             -X PATCH -f body="$RESOLVED_BODY"

--- a/.github/actions/apply/lib/post-comment.sh
+++ b/.github/actions/apply/lib/post-comment.sh
@@ -25,6 +25,30 @@ set -euo pipefail
 # `gh pr comment --body-file` and `gh api -f key=@file` both stream from
 # the file instead of expanding it into the command line.
 
+# GitHub's comment API caps a single body at 65,536 characters ("Body is
+# too long"). Streaming from a file dodges ARG_MAX but *not* this ceiling,
+# so truncate an over-long body here — defence in depth for every comment
+# the action posts (e.g. a large module's a11y report). The budget is in
+# bytes; a UTF-8 string's byte length is >= its character count, so keeping
+# under the byte budget guarantees we're under the character cap, with
+# headroom for the truncation notice. We keep the head of the body (the
+# marker + summary live at the top) and cut on a line boundary.
+MAX_BYTES="${COMMENT_MAX_BYTES:-60000}"
+BODY_BYTES=$(wc -c < "$BODY_FILE")
+if [ "$BODY_BYTES" -gt "$MAX_BYTES" ]; then
+  NOTICE=$'\n\n---\n> ⚠️ This comment was truncated because it exceeded GitHub'\''s 65,536-character limit. See the full report in the render branch / workflow artifacts.'
+  NOTICE_BYTES=$(printf '%s' "$NOTICE" | wc -c)
+  BUDGET=$((MAX_BYTES - NOTICE_BYTES))
+  TRUNCATED=$(mktemp)
+  # Cut to the byte budget, then drop the final (possibly partial) line so
+  # we always end on a clean line boundary and never leave a half-written
+  # multi-byte character.
+  head -c "$BUDGET" "$BODY_FILE" | sed '$d' > "$TRUNCATED"
+  printf '%s\n' "$NOTICE" >> "$TRUNCATED"
+  echo "post-comment: body was ${BODY_BYTES} bytes; truncated to fit GitHub's 65,536-char comment limit." >&2
+  BODY_FILE="$TRUNCATED"
+fi
+
 COMMENT_ID=$(gh api \
   "repos/${REPO}/issues/${PR_NUMBER}/comments" \
   --paginate \

